gnocchi/storage/_carbonara.py

```diff
@@ -117,9 +118,15 @@ def _store_metric_measures(metric, timestamp_key, aggregation,
                                granularity, data, offset=None, version=3):
         raise NotImplementedError
 
-    @staticmethod
-    def _list_split_keys_for_metric(metric, aggregation, granularity,
+    def _list_split_keys_for_metric(self, metric, aggregation, granularity,
                                     version=3):
+        return set(map(
+            functools.partial(carbonara.SplitKey, sampling=granularity),
+            utils.to_timestamps(self._list_split_keys(
```

Comment:
We'll never remove the test change because we want to pass SplitKey objects with numpy types in it.

That's why the patch is big in the end. We change the typo in the tests.